pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/state/PulsarMetadataStateStoreImpl.java

```diff
@@ -111,6 +112,20 @@ public CompletableFuture<ByteBuffer> getAsync(String key) {
                                 .orElse(null));
     }
 
+    @Override
+    public StateValue getStateValue(String key) {
+        return getStateValueAsync(key).join();
+    }
+
+    @Override
+    public CompletableFuture<StateValue> getStateValueAsync(String key) {
+        return store.get(getPath(key))
+                .thenApply(optRes ->
+                        optRes.map(x ->
+                            new StateValue(ByteBuffer.wrap(x.getValue()), x.getStat().getVersion(), null))
```

Comment:
Is there a benefit of `ByteBuffer.wrap` when it's immediately accessed directly in `StateValue` constructor (in the questionable way)? Perhaps it would be better to pass what `byte[]` directly?

Reply:
Great, I will use `byte[]`